Sources/ArgumentParser/Usage/DumpHelpGenerator.swift

```diff
@@ -9,15 +9,12 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if swift(>=5.11)
+#if swift(>=6.0)
 internal import ArgumentParserToolInfo
 internal import class Foundation.JSONEncoder
-#elseif swift(>=5.10)
+#else
 import ArgumentParserToolInfo
```

Comment:
I'm slightly concerned about this. The `ArgumentParserToolInfo` library is statically built, and therefore not distributed. However, this is going to serialise the dependency information into the swiftmodule. This might therefore break builds with swift <6.0 on Windows. I do have a pending change (https://github.com/swiftlang/swift/pull/63814) that would help with this situation, but I was running into some trouble with Linux builds ...

Reply:
Hmm implementation only import causes real miscompiles with library evolution so I think it really really needs to go.

Reply:
@rauhul Where are we seeing this issue show up right now? Can we gate this change so that the attribute is still used on Windows/Linux?

Reply:
Theres a ton of context in this radar: rdar://78129903 (`@_implementationOnly` should either work in non-library-evolution-enabled modules, or be disallowed because we know it doesn't work) (sorry non-apple folks).

The jist of it is that this attribute has never properly worked without library evolution and introduces very hard to locate miscompiles due to different compilation units disagreeing about the layout of types.

@mikeash thoughts?

Reply:
I guess I don't understand the benefits of using this attribute at all for non library evo code

Reply:
I'm not sure myself! Looking at @compnerd's PR, it might break builds by emitting different autolinking info?